src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/multiblocks/ArmorForge.java

```diff
@@ -85,12 +95,13 @@ private void craft(Player p, ItemStack output, Inventory inv, Block dispenser) {
             for (int j = 0; j < 4; j++) {
                 int current = j;
 
+                ItemStack finalOutput = output;
                 Slimefun.runSync(() -> {
                     if (current < 3) {
                         p.getWorld().playSound(p.getLocation(), Sound.BLOCK_ANVIL_USE, 1F, 2F);
                     } else {
                         p.getWorld().playSound(p.getLocation(), Sound.ENTITY_ARROW_HIT_PLAYER, 1F, 1F);
-                        handleCraftedItem(output, dispenser, inv);
+                        handleCraftedItem(finalOutput, dispenser, inv);
```

Comment:
I don't see the reason to create a separate variable called finalOutput here, am I overlooking something?

Reply:
you can't use reassigned variables in lambda expressions in java

Reply:
okay good to know